pyvrp/Model.py

```diff
@@ -180,6 +180,9 @@ def add_edge(
             """
             warn(msg, ScalingWarning)
 
+        if frm == to and (distance != 0 or duration != 0):
```

Comment:
I considered disallowing self-loops at all, however this way we avoid the need for an if statement when adding all edges from a matrix using a nested for-loop, as the self-loop can be added as long as it's 0 valued.